app/uk/gov/hmrc/minorentityidentificationfrontend/controllers/uaControllers/CaptureOfficePostcodeController.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.minorentityidentificationfrontend.controllers.uaControllers
+
+import play.api.mvc._
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals.internalId
+import uk.gov.hmrc.auth.core.{AuthConnector, AuthorisedFunctions}
+import uk.gov.hmrc.http.InternalServerException
+import uk.gov.hmrc.minorentityidentificationfrontend.config.AppConfig
+import uk.gov.hmrc.minorentityidentificationfrontend.featureswitch.core.config.{EnableFullUAJourney, FeatureSwitching}
+import uk.gov.hmrc.minorentityidentificationfrontend.forms.uaForms.CaptureOfficePostcodeForm
+import uk.gov.hmrc.minorentityidentificationfrontend.services.{JourneyService, StorageService}
+import uk.gov.hmrc.minorentityidentificationfrontend.views.html.uaViews.capture_office_postcode_page
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class CaptureOfficePostcodeController @Inject()(mcc: MessagesControllerComponents,
+                                                view: capture_office_postcode_page,
+                                                journeyService: JourneyService,
+                                                storageService: StorageService,
+                                                val authConnector: AuthConnector
+                                           )(implicit val config: AppConfig, executionContext: ExecutionContext)
+  extends FrontendController(mcc) with AuthorisedFunctions with FeatureSwitching {
+
+  def show(journeyId: String): Action[AnyContent] = Action.async {
+    implicit request =>
+      authorised().retrieve(internalId) {
+          case Some(authInternalId) =>
+            if(isEnabled(EnableFullUAJourney)) {
```

Comment:
Roberto has suggested we try to use programmatic routing to control access to the UA flow. However, I see Tadas is also using this approach so we may as well continue.